tests/test_fmi.py

```diff
@@ -339,6 +445,14 @@ def test_log_file_name(self):
         path, file_name = os.path.split(full_path)
         assert model.get_log_file_name() == file_name.replace(".","_")[:-4]+"_log.txt"
 
+    @testattr(stddist = True)
+    def test_unzipped_fmu_exceptions(self):
+        """ Verify exception is raised if 'fmu' is a file and allow_unzipped_fmu is set to True, with FMUModelCS2. """
+        err_msg = "Argument named 'fmu' must be a directory if argument 'allow_unzipped_fmu' is set to True."
+        full_path = os.path.join(file_path, "files", "FMUs", "XML", "CS2.0")
+        with nose.tools.assert_raises_regex(FMUException, err_msg):
```

Comment:
Also here, why a with?

Reply:
Same as above, it will also catch scenariso where no exception is thrown with minimal code. I guess the other way would be:
`
exception_raised = False
try:
  foo()
except FMUException as e:
  exception_raised = True
  <verify exception message>
assert exception_raised, "helpful error message"
`
but is a lot more code to achieve the same thing.

Reply:
Why not just nose.tools.assert_raises(FMUException, function, arguments)?

Reply:
See my reply in the other comment regarding the context managers, I think the difference is minor but we also test the error message.

Reply:
Also here. OK but I think context managers make it harder to read what is being tested.